### PR TITLE
feat: Add roles to conversation entity

### PR DESCRIPTION
**Pull request by lipis**

### CODE REVIEW

File: src/script/conversation/ConversationRepository.js [hunk 1]
```diff
@@ -485,7 +485,9 @@ export class ConversationRepository {
         const data = this.conversationMapper.mergeConversation(localConversations, remoteConversations);
         return this.conversation_service.save_conversations_in_db(data);
       })
-      .then(conversationsData => this.mapConversations(conversationsData))
+      .then(conversationsData => {
+        return this.mapConversations(conversationsData);
```
Comment (bennycode) [inline src/script/conversation/ConversationRepository.js]: can become a one-liner

File: src/script/conversation/ConversationRepository.js [hunk 2]
```diff
@@ -815,7 +817,7 @@ export class ConversationRepository {
 
   isUserGroupAdmin(conversationEntity, userEntity) {
     // TODO: once we have the roles from the backend, use these
```
Comment (bennycode) [inline src/script/conversation/ConversationRepository.js]: TODO can now be removed

File: src/script/entity/Conversation.js [hunk 3]
```diff
@@ -76,6 +76,7 @@ export class Conversation {
     this.participating_user_ets = ko.observableArray([]); // Does not include self user
     this.participating_user_ids = ko.observableArray([]); // Does not include self user
     this.selfUser = ko.observable();
+    this.roles = {wire_admin: [], wire_member: []};
```
Comment (bennycode) [inline src/script/entity/Conversation.js]: please make a constant for "wire_admin" and "wire_member"

File: src/script/conversation/ConversationMapper.js [hunk 4]
```diff
@@ -352,11 +353,30 @@ export class ConversationMapper {
...
         name,
         receipt_mode,
+        roles: {},
         status: selfState.status,
         team_id: team,
         type,
       };
 
+      // Add roles for self
+      if (selfState.conversation_role) {
```
Comment (ffflorian) [inline src/script/conversation/ConversationMapper.js:363]:
```suggestion
      if (selfState.conversation_role && !(selfState.conversation_role in updates.roles)) {
```

File: src/script/conversation/ConversationMapper.js [hunk 5]
```diff
@@ -352,11 +353,30 @@ export class ConversationMapper {
...
+        if (!(selfState.conversation_role in updates.roles)) {
```
Comment (ffflorian) [inline src/script/conversation/ConversationMapper.js:364]:
```suggestion
```

File: src/script/conversation/ConversationMapper.js [hunk 6]
```diff
@@ -352,11 +353,30 @@ export class ConversationMapper {
...
+          updates.roles[selfState.conversation_role] = [];
+        }
```
Comment (ffflorian) [inline src/script/conversation/ConversationMapper.js]:
```suggestion
```

File: src/script/conversation/ConversationMapper.js [hunk 7]
```diff
@@ -352,11 +353,28 @@ export class ConversationMapper {
...
         receipt_mode,
+        roles: {},
         status: selfState.status,
         team_id: team,
         type,
       };
 
+      // Add roles for self
+      if (selfState.conversation_role && !(selfState.conversation_role in updates.roles)) {
+          updates.roles[selfState.conversation_role] = [];
```
Comment (lipis) [inline src/script/conversation/ConversationMapper.js:365]:
```suggestion
        updates.roles[selfState.conversation_role] = [];
```

**Pull request merged**